### PR TITLE
MAISTRA-2426 rework federation status api

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,11 +6,12 @@ require (
...
 	github.com/gogo/protobuf v1.3.1
 	github.com/google/go-cmp v0.5.2
-	google.golang.org/grpc v1.27.0
-	k8s.io/api v0.19.3
-	k8s.io/apimachinery v0.19.3
-	k8s.io/client-go v0.19.3
-	k8s.io/code-generator v0.19.3
+	github.com/maistra/xns-informer v0.0.0-20210602150316-33463c1f7d6d
+	google.golang.org/grpc v1.28.1
+	k8s.io/api v0.19.2
```
Comment (jwendell) [inline go.mod]: Is this downgrade necessary? Will it impact istio's go.mod/sum?

Comment (rcernich) [reply go.mod]: These changes are consistent with what's in xns-informer, which is why I made them.

Comment (jwendell) [reply go.mod]: I had forgot we bumped maistra/istio to 1.9. Which means everything should bump to 0.20.x (version that istio uses).

Comment (rcernich) [reply go.mod]: I'll update this once maistra/xns-informer#14 is merged

Comment (rcernich) [reply go.mod]: I've rebased and updated all of these to v0.20.1, same as xns-informer.

File: Makefile [hunk 2]
```diff
@@ -68,6 +70,10 @@ generate-client:
 	rm -rf client && mv maistra.io/api/client . && rm -rf maistra.io
 	## Hack - Because we are using core, client-gen hardcodes it to /api
 	find client -name core_client.go -exec sed -i 's|config.APIPath = "/api"|config.APIPath = "/apis"|' {} \;
+	## Hack - Because we are not using +groupName=maistra.io in doc.go, the incorrect group is used in fake types
```
Comment (jwendell) [inline Makefile:73]: Does this mean this hack would go away if we add `+groupName=maistra.io` in the right place? If so, do you want to add a commit that does it so we don't need this?

Comment (rcernich) [reply Makefile:73]: Adding the groupName changes the generated interfaces, which would require changes to maistra/istio.  We should do that as part of another PR/issue, but I would consider that low priority at this point.

**Review by jwendell — APPROVED** [review]

**Pull request merged**